File: app/src/app/core/services/api/models/oseba.ts
import { Id } from './id'
export interface Oseba {
	'_id': Id<Oseba>
	geslo: string
	ime: string
	letnik: number
	priimek: string
	tip: Array<'UCENEC' | 'UCITELJ' | 'INSTRUKTOR' | 'ADMIN' | 'KONTAKT' | 'SERVER'>
	username: string
}

File: api/core/src/main/kotlin/core/domain/Oseba.kt
package core.domain

import core.base.Hashed
import core.base.Id
import kotlinx.serialization.Contextual
import kotlinx.serialization.Serializable

@Serializable
data class Oseba(
    override var _id: Id<Oseba> = Id(),
    var tip: MutableSet<Tip>,
    val geslo: Hashed,
    var letnik: Int,
    @Contextual val username: core.base.Encrypted,
    @Contextual val ime: core.base.Encrypted,
    @Contextual val priimek: core.base.Encrypted,
) : Entiteta<Oseba> {
    enum class Tip { UCENEC, UCITELJ, INSTRUKTOR, ADMIN, KONTAKT, SERVER }
}
